### PR TITLE
Linear Regression out of preview

### PR DESCRIPTION
**Pull request by ethanglaser**
# Description
Migration of sklearnex LinearRegression out of preview.

Overview of changes
- Combined former _common and linear files into single linear.py to match general layout of other non-preview sklearnex algos
- Migration of general test _preview_linear into test_linear
- Removal of preview linear_model and updates to any mentions (in dispatcher, test_monkeypatch, test_memory_usage)

### CODE REVIEW

**Comment by ethanglaser** [comment]
Will need to take a look at GPU sklearnex validation on 3.8 before merging

File: sklearnex/tests/test_memory_usage.py [hunk 1]
```diff
@@ -109,7 +109,7 @@ def remove_duplicated_estimators(estimators_list):
 )
 estimators = [
     PreviewPCA,
-    PreviewLinearRegression,
+    LinearRegression,
```
Comment (Alexsandruss) [inline sklearnex/tests/test_memory_usage.py:112]: All non-preview patched estimators are added by default
```suggestion
```

File: sklearnex/tests/test_memory_usage.py [hunk 2]
```diff
@@ -28,6 +28,7 @@
 from sklearn.model_selection import KFold
 
 from sklearnex import get_patch_map
+from sklearnex.linear_model import LinearRegression
```
Comment (Alexsandruss) [inline sklearnex/tests/test_memory_usage.py:31]:
```suggestion
```

**Comment by samir-nasibli** [comment]
@ethanglaser  On #1182

We deselected some test cases actually for preview functionality as I see on this [PR](https://github.com/intel/scikit-learn-intelex/pull/1182/files):
https://github.com/intel/scikit-learn-intelex/blame/master/deselected_tests.yaml#L242:L251
Please check if possible to enable this tests

Superficially looks good to me.

File: sklearnex/tests/test_memory_usage.py (hunk 1 above)
Comment (ethanglaser) [reply sklearnex/tests/test_memory_usage.py:112]: Thanks for clarification

**Comment by ethanglaser** [comment]
> @ethanglaser On #1182
> 
> We deselected some test cases actually for preview functionality as I see on this [PR](https://github.com/intel/scikit-learn-intelex/pull/1182/files): https://github.com/intel/scikit-learn-intelex/blame/master/deselected_tests.yaml#L242:L251 Please check if possible to enable this tests
> 
> Superficially looks good to me.

Thanks for mentioning - I will test these out

**Comment by ethanglaser** [comment]
Current CI without 3.8 pvc sklearnex step: http://intel-ci.intel.com/ee7f3d76-7f23-f135-a4fa-a4bf010d0e2e

**Review by icfaust — APPROVED** [review]

**Comment by ethanglaser** [comment]
CI after rebase: http://intel-ci.intel.com/ee8252f7-55ab-f1dc-8b5c-a4bf010d0e2e

**Pull request merged**